### PR TITLE
Add an engine PR template

### PR DESCRIPTION
**Pull request by xster**
Mostly based on the framework template https://github.com/flutter/flutter/blob/master/.github/PULL_REQUEST_TEMPLATE.md

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+I added the following tests:
+
+*Replace this with a list of the tests that you added as part of this PR. A
+change in behaviour with no test covering it will likely get reverted
+accidentally sooner or later. PRs must include tests for all
+changed/updated/fixed behaviors. See [testing the engine] for instructions on
+writing and running engine tests.*
+
+## Checklist
```
Comment (chinmaygarde) [inline .github/PULL_REQUEST_TEMPLATE.md:23]: This is not part of the commit message right? I don't think having the same completed checklist over and over in the commit log adds value to the log. If its just part of the GitHub comment, then thats fine.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Breaking Change
+
+Did any tests fail when you ran them? Please read [Handling breaking changes].
+
+- [ ] No, no existing tests failed, so this is *not* a breaking change.
+- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
+   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
+   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
+   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*
```
Comment (chinmaygarde) [inline .github/PULL_REQUEST_TEMPLATE.md:43]: This guidance has seldom proven applicable for engine changes. How about just mentioning changes to the public interfaces in the embeddings?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (xster) [reply .github/PULL_REQUEST_TEMPLATE.md:23]: No. The squash commit message is unrelated to the pull request comment history.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (xster) [reply .github/PULL_REQUEST_TEMPLATE.md:43]: I agree it's less common. Public embedding API would be breaking changes. But I'm open to removing this section if you think it's rare enough for the engine repo.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (chinmaygarde) [reply .github/PULL_REQUEST_TEMPLATE.md:23]: Sounds good.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (chinmaygarde) [reply .github/PULL_REQUEST_TEMPLATE.md:43]: Don't have a strong opinion either way. Lets land this and tweak as necessary.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+<!-- Links -->
+[issue database]: https://github.com/flutter/flutter/issues
+[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
+[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
+[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
+[CLA]: https://cla.developers.google.com/
+[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
+[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
```
Comment (gaaclarke) [inline .github/PULL_REQUEST_TEMPLATE.md:52]: Let's make these links follow the same capitalization scheme.

Comment (xster) [reply .github/PULL_REQUEST_TEMPLATE.md]: oh good point. Made them all lowercase, since they mostly don't even match the title of the wikis
